registry/handlers/app.go

```diff
@@ -211,11 +215,72 @@ func NewApp(ctx context.Context, configuration configuration.Configuration) *App
 // process. Because the configuration and app are tightly coupled,
 // implementing this properly will require a refactor. This method may panic
 // if called twice in the same process.
-func (app *App) RegisterHealthChecks() {
-	health.RegisterPeriodicThresholdFunc("storagedriver_"+app.Config.Storage.Type(), 10*time.Second, 3, func() error {
-		_, err := app.driver.List(app, "/") // "/" should always exist
-		return err                          // any error will be treated as failure
-	})
+func (app *App) RegisterHealthChecks(healthRegistry *health.Registry) {
```

Comment:
That's an interesting idea, but to me it implies that the function can register the checks in multiple health registries. While we could make it support that, I can't see a use case where it would make sense. So I'm not sure this is a good fit for the situation.